services/cd-service/pkg/repository/transformer.go

```diff
@@ -547,43 +547,28 @@ func writeCommitData(ctx context.Context, sourceCommitId string, sourceMessage s
 	if err := util.WriteFile(fs, fs.Join(commitAppDir, ".gitkeep"), make([]byte, 0), 0666); err != nil {
 		return GetCreateReleaseGeneralFailure(err)
 	}
-	err := writeNewReleaseEvent(ctx, eventId, sourceCommitId, fs, environments)
+	envMap := make(map[string]struct{}, len(environments))
+	for _, env := range environments {
+		envMap[env] = struct{}{}
+	}
+	err := writeEvent(eventId, sourceCommitId, fs, &event.NewRelease{
+		Environments: envMap,
+	})
 	if err != nil {
 		return fmt.Errorf("error while writing event: %v", err)
 	}
 	return nil
 }
 
-func writeNewReleaseEvent(ctx context.Context, eventId string, sourceCommitId string, filesystem billy.Filesystem, envs []string) error {
+func writeEvent(
+	eventId string,
+	sourceCommitId string,
+	filesystem billy.Filesystem,
+	ev event.Event,
+) error {
 	eventDir := commitEventDir(filesystem, sourceCommitId, eventId)
-	_, err := filesystem.Stat(eventDir)
-	if err != nil {
-		if !errors.Is(err, fs.ErrNotExist) {
-			// anything except "not exists" is an error:
-			return fmt.Errorf("event directory %s already existed: %v", eventDir, err)
-		}
-	}
-	if err := filesystem.MkdirAll(eventDir, 0777); err != nil {
-		return fmt.Errorf("could not create directory %s: %v", eventDir, err)
-	}
-	for i := range envs {
-		env := envs[i]
-		environmentDir := filesystem.Join(eventDir, "environments", env)
-		if err := filesystem.MkdirAll(environmentDir, 0777); err != nil {
-			return fmt.Errorf("could not create directory %s: %v", environmentDir, err)
-		}
-		environmentNamePath := filesystem.Join(environmentDir, ".gitkeep")
-		if err := util.WriteFile(filesystem, environmentNamePath, make([]byte, 0), 0666); err != nil {
-			return fmt.Errorf("could not write file %s: %v", environmentNamePath, err)
-		}
-	}
-	eventTypePath := filesystem.Join(eventDir, "eventType")
-	if err := util.WriteFile(filesystem, eventTypePath, []byte(event.NewReleaseEventName), 0666); err != nil {
-		return fmt.Errorf("could not write file %s: %v", eventTypePath, err)
-	}
-
 	// Note: we do not store the "createAt" date here, because we use UUIDs with timestamp information
```

Comment:
either move this comment somewhere else, or delete it, if it doesn't fit anymore.